patterns/1-initial/extensions-to-manage-contributions-at-scale.md

```diff
@@ -0,0 +1,81 @@
+3. <b>Dependency management:</b>  Each extension should be careful to pin the version range of the primary repository that it is built against (the same way it would any other dependency) and should be careful in its use of other dependencies that shadow dependencies of the primary repository such that the versions it chooses for those dependencies are compatible with the primary repository versions selected. Any conflicts with primary repository will be caught in the test framework for the extension.
+4. <b>Testing strategy:</b> How to test extensions both individually and in combination?
+   - <b>Testing extension individually:</b> Extensions template will provide a test framework to be used by the extension developers to test the capability added. This can include a framework for unit tests, runtime performance and quality tests.
+   - <b>Testing extension in combination with primary repository:</b> Extension developers have a well-patterned method for testing their extension against specific versions of the primary repository without involvement from the primary repository's maintainers.
+   - <b>Testing extension in combination with other extensions:</b> Providing a test framework for this scenario could result in being excessive especially if there are a large number of extensions that are still being explored by users and unlikely to be all used in combination. If a user runs into conflicts while using extensions in combination (which should be unlikely with sufficient loose coupling), the user can raise an issue to the respective extension owners who will sort it out. As an extension reaches later phases of the lifecycle and gets merged into the primary repository, it would be tested in combination with rest of library and any dependency conflicts will have to be resolved at the time.
+5. <b>Discoverability and Usability:</b>
+   - Make extensions easily discoverable with a publishing page showing the extensions that users have created and want to share for product usage.
+   - Allow registration of extensions with the primary project for users to leverage extensions alongside the original project, thus keeping the same user experience.
+6. <b>Lifecycle of extensions and maintainability:</b> Establish the lifecycle for extensions from creation to porting into the primary codebase, along with clear ownership guidelines.
+   - Extension creators continue maintaining the extension, providing any support and fixing defects. Any extension left unmaintained will be unlisted from the publishing page.
+   - Create criteria for when an extension can be ported to the primary repository, such as adoption of the extension by internal products and demand for the feature.
+   - Porting process of the extension to the primary repository will follow more stringent code review guidelines as set by library maintainers.
+
+Following the principles outlined above ensures that:
```

Comment:
I wonder if the points below here should rather go into "Resulting Context".
What do you think @NewMexicoKid?

Reply:
ya I was also doubtful where to put them, as we used the points to emphasize why we need these principles

Reply:
I would be happy to keep it here for now, and possibly rework it into the Resulting Context or elsewhere in future PRs/reviews.

Reply:
@spier -- I agree that the points (lines 59-63) would probably be best represented in the Resulting Context section.